wp_api/src/login/url_discovery.rs

```diff
@@ -129,6 +109,24 @@ impl AutoDiscoveryResult {
     ) -> Option<&AutoDiscoveryAttemptResult> {
         self.attempts.get(attempt_type)
     }
+
+    /// Returns either the successful attempt or the most relevant failure
+    pub fn combined_result(
+        &self,
+    ) -> Result<&AutoDiscoveryAttemptSuccess, &AutoDiscoveryAttemptFailure> {
+        if let Some(success) = self.find_successful() {
+            success.api_discovery_result.as_ref()
+        } else {
+            Err(self
+                .attempts
+                .iter()
+                .flat_map(|(_, a)| a.api_discovery_result.as_ref().err())
+                // Sort in descending order so the most important error is returned
+                .sorted_by(|a, b| b.compare_importance(a))
+                .next()
+                .expect("If the discovery was unsuccessful, there is at least one error"))
```

Comment:
Nitpick: I think you can get rid of this unnecessary `expect` if deleting `find_successful` and re-implementing it within this function.